app/Main.hs

```diff
@@ -188,32 +198,69 @@ eventProcessor submit nextEvent = go
             IssueFanout -> submit NodeCommand.Fanout >> go
             InitHead period -> submit (NodeCommand.Init period) >> go
             CommitToHead utxoToCommit -> submit (NodeCommand.Commit utxoToCommit) >> go
-            Bet _txIn _pp -> liftIO (putStrLn "no betting yet") >> go
+            Bet txIn txInLovelace pp -> do
+              state <- ask
+              let unsignedTx = either error id $ buildBetTx txIn txInLovelace state pp
+                  signedTx = mkSignedTx state unsignedTx
+              submit (NodeCommand.newTx signedTx)
+              go
             Claim collateralTxIn cp -> do
               state <- ask
               let unsignedTx = either error id $ buildClaimTx collateralTxIn state cp
-                  signedTx = signTx state.hsUserCredentials.userSkey unsignedTx
+                  signedTx = mkSignedTx state unsignedTx
               submit (NodeCommand.newTx signedTx) >> go
 
+mkSignedTx :: HeadState -> TxBody AlonzoEra -> Tx AlonzoEra
+mkSignedTx state = signTx state.hsUserCredentials.userSkey
+
 betConstant :: Lovelace
 betConstant = 10000000
 
+{-# ANN gameDatum "HLint: ignore Eta reduce" #-}
+gameDatum :: UserInput.PlayParams -> Ledger.PubKeyHash -> GameDatum
```

Comment:
Maybe we could name this something like `buildDatum`? Also, why add the hlint ignore annotation?

Reply:
By default it will match any pattern `f a b ... z = g ... z` and suggest to drop (eta reduce) the `z` away. We could make a case to disable eta reduction suggestions globally, as one could argue code is "clearer" without it, but it is a very common pattern in haskell (and I think part of our style guide) which I think is a good reason to have it enabled and apply it consistently

Reply:
I think, we can avoid this pattern and actually do make it better, though slightly more verbose, like this:
```
buildDatum :: UserInput.PlayParams -> Ledger.PubKeyHash -> GameDatum
buildDatum playParams pkh =
  GameDatum
    { gdGesture = encryptGesture playParams.ppGesture playParams.ppSalt
    , gdPkh = pkh
    }
```